clarify supported openai models in copilot (#7202)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/dbt-copilot-faqs.md

With automatic code generation, let <Constant name="copilot" /> [generate code](/docs/cloud/use-dbt-copilot) using natural language, and [generate documentation](/docs/build/documentation), [tests](/docs/build/data-tests), [metrics](/docs/build/metrics-overview), and [semantic models](/docs/build/semantic-models) for you with the click of a button in the [<Constant name="cloud_ide" />](/docs/cloud/dbt-cloud-ide/develop-copilot) and, now in beta, [<Constant name="visual_editor" />](/docs/cloud/use-visual-editor).

> dbt Copilot is optimized for OpenAI's gpt-3.x, gpt-4o, gpt-4.1-[mini|nano], gpt-4.5 (deprecated by OpenAI) models. Using other models (like o1 and o2) can affect performance and accuracy, and functionality with other models isn't guaranteed.

Author: ```suggestion
<Constant name="copilot"/> is optimized for OpenAI's gpt-3.x, gpt-4o, gpt-4.1-[mini|nano], gpt-4.5 (deprecated by OpenAI) models. Other models, like o1 and o2, are not supported and will not work with <Constant name="copilot"/>.
```

With automatic code generation, let <Constant name="copilot" /> [generate code](/docs/cloud/use-dbt-copilot) using natural language, and [generate documentation](/docs/build/documentation), [tests](/docs/build/data-tests), [metrics](/docs/build/metrics-overview), and [semantic models](/docs/build/semantic-models) for you with the click of a button in the [<Constant name="cloud_ide" />](/docs/cloud/dbt-cloud-ide/develop-copilot) and, now in beta, [<Constant name="visual_editor" />](/docs/cloud/use-visual-editor).

> <Constant name="copilot"/> is optimized for OpenAI's gpt-3.x, gpt-4o, gpt-4.1-[mini|nano], gpt-4.5 (deprecated by OpenAI) models. Other models, like o1 and o2, are not supported and will not work with <Constant name="copilot"/>.

Author: not sure if i should add backticks to the `gpt-3.x`, `gpt-4o`etc. or leave as is?

> <Constant name="copilot"/> is optimized for OpenAI's gpt-3.x, gpt-4o, gpt-4.1-[mini|nano], gpt-4.5 (deprecated by OpenAI) models. Other models, like o1 and o2, are not supported and will not work with <Constant name="copilot"/>.

Reviewer: ```suggestion
<Constant name="copilot"/> is optimized for OpenAI's gpt-3.x, gpt-4o, gpt-4.1-[mini|nano], and gpt-4.5 (deprecated by OpenAI) models. Other models, like o1 and o2, are not supported and will not work with <Constant name="copilot"/>.
```

> <Constant name="copilot"/> is optimized for OpenAI's gpt-3.x, gpt-4o, gpt-4.1-[mini|nano], and gpt-4.5 (deprecated by OpenAI) models. Other models, like o1 and o2, are not supported and will not work with <Constant name="copilot"/>.

Author: ```suggestion
<Constant name="copilot"/> is optimized for OpenAI's `gpt-3.x`, `gpt-4o`, `gpt-4.1-[mini|nano]`, and `gpt-4.5` (deprecated by OpenAI) models. Other models, like `o1` and `o2`, are not supported and will not work with <Constant name="copilot"/>.
```
